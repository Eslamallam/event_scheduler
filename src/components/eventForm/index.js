import React from "react";
import { useForm } from "../../helpers/useForm";
import { validate } from "../../helpers/validate";
import * as actions from "../../store/actions";
import { useDispatch } from "react-redux";
import { AboutSection } from "./AboutSection";
import { CoordinatorSection } from "./CoordinatorSection";
import { WhenSection } from "./whenSection";
import { SubmitSection } from "./SubmitSection";
import { SuccessSection } from "./SuccessSection";

export const EventForm = () => {
  const {
    handleChange,
    handleSubmit,
    values,
    errors,
    chars,
    maxChars,
    showSuccess,
  } = useForm(formSubmit, validate);

  const dispatch = useDispatch();

  function formSubmit() {
    const formData = {};

    // data to be submitted
    formData.title = values.title;
    formData.description = values.description;
    formData.category_id = parseInt(values.category_id) || null;
    formData.paid_event = values.event_type === "paid" ? true : false;
    formData.event_fee = parseInt(values.event_fee) || null;
    formData.reward = parseInt(values.reward) || null;
    formData.date = values.date + "T" + values.event_time;
    formData.duration = parseInt(values.duration) * 60 * 60 || null;
    formData.coordinator = {
      email: values.coordinator.split("-")[0],
      id: values.coordinator.split("-")[1],
    };

    console.log(formData);
    dispatch(actions.doCreateEvent(formData));
  }

  return (
    <div className="event">
      {showSuccess ? (
        <SuccessSection />
      ) : (
        <form onSubmit={handleSubmit} noValidate>
          <AboutSection
            handleChange={handleChange}
            inputValues={values}
            errors={errors}
            chars={chars}
            maxChars={maxChars}
          />
          <CoordinatorSection
            handleChange={handleChange}
            inputValues={values}
            errors={errors}
          />
          <WhenSection
            handleChange={handleChange}
            inputValues={values}
            errors={errors}
          />

          <SubmitSection />
        </form>
      )}
    </div>
  );
};
