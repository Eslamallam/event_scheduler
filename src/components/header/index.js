import React from "react";

export const Header = () => {
  return (
    <div className="header">
      <h2>New event</h2>
    </div>
  );
};
