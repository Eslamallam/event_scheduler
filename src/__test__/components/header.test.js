import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJSON from "enzyme-to-json";
import { Header } from "../../components/header";

configure({ adapter: new Adapter() });

test("should render Header correctly", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper.find("h2").text()).toBe("New event");
  expect(toJSON(wrapper)).toMatchSnapshot();
});
