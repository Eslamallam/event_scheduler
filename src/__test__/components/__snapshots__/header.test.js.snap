// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className="header"
>
  <h2>
    New event
  </h2>
</div>
`;
